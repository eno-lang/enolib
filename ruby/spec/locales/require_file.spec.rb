# frozen_string_literal: true

locales = ['De', 'Es']

# This can/should occasionally be commented out (and back in afterwards)
# to reaffirm the additional locales are not required through some other file
require 'enolib/locales'

describe 'Locales required through require file' do
  locales.each do |locale_code|
    describe locale_code do
      it 'provides a working locale' do
        errored = false

        begin
          Enolib.parse(':invalid', locale: Enolib::Locales.const_get(locale_code))
        rescue Enolib::ParseError => error
          expect(error.message).to match_snapshot
          errored = true
        end

        expect(errored).to be true
      end
    end
  end
end
