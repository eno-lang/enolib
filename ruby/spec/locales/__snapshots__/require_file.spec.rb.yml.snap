---
"./spec/locales/require_file.spec.rb[1:1:1]": |-
  Zeile 1 folgt keinem spezifierten Muster.

    Zeile | Inhalt
   >    1 | :invalid
"./spec/locales/require_file.spec.rb[1:2:1]": |-
  Línea 1 no sigue un patrón especificado.

    Línea | Contenido
   >    1 | :invalid
